### ISSUE TITLE
AddProvider for custom tag doesn't work with some custom types

### ISSUE THREAD

**Issue by andrew-werdna** [bxcodec/faker#101]
### Example struct field and type:   
```
import "github.com/google/uuid"
...
type Product struct {
  ProductID uuid.UUID `faker:"google_uuid"`
}
...
_ = faker.AddProvider("google_uuid", func(v reflect.Value) (interface{}, error) {
  return uuid.New(), nil // to instantiate a new, random instance
})
```   
### Result:   
```
panic: reflect.MakeSlice of non-slice type [recovered]
    panic: reflect.MakeSlice of non-slice type
```

### What appears to be the problem:   
If you look, google's UUID is defined as `type UUID [16]byte`.   
It would appear that in `setDataWithTag()` the `reflect.Indirect(v)` is returning `array`, but if you were to do `v.Kind()` you would get `uuid.UUID`. In the switch statement you would arrive at the `reflect.Slice, reflect.Array`, call `userDefinedArray()`, and then arrive at the following [line](https://github.com/bxcodec/faker/blob/9169c33ae9926e5b8f8732909790ee20b10b736a/faker.go#L625), where the panic occurs 
```
array := reflect.MakeSlice(v.Type(), len, len)
```   

So even despite adding the custom tag with `AddProvider()` and including the ability to create a new, random UUID, it never checks if there is a custom tag before going into `userDefinedArray()`

### Proposal
Allow a user-defined tag (added with `AddProvider()`), to take precedence, and if the user provides that ability, check it first before defaulting to `userDefinedArray()`.

**Comment by andrew-werdna** [comment]
I am about to open a PR that if acceptable to you will fix this particular problem. If you are okay with it, this could potentially be extended to work with other types (other than just `reflect.Array, reflect.Slice`)

**Comment by bxcodec** [comment]
fixed by #102
